pdns/pdns_recursor.cc

```diff
@@ -1118,6 +1118,13 @@ static void startDoResolve(void *p)
     // Check if the query has a policy attached to it
     if (wantsRPZ) {
       appliedPolicy = luaconfsLocal->dfe.getQueryPolicy(dc->d_mdp.d_qname, dc->d_source, sr.d_discardedPolicies);
+
+      if(appliedPolicy.d_kind != DNSFilterEngine::PolicyKind::NoAction) {
+        if(t_rpzremotes)
```

Comment:
explicitly testing for `nullptr` makes the code easier to read. And can you please use accolades?

```cpp
if (t_rpzremotes != nullptr) {
  t_rpzremotes->push_back(dc->d_source);
}
```